neo4j-jdbc/src/main/java/org/neo4j/jdbc/Neo4jStatement.java

```diff
@@ -162,7 +162,15 @@ protected void checkClosed() throws SQLException {
 	 */
 	@Override public boolean getMoreResults() throws SQLException {
 		this.checkClosed();
-		return this.currentResultSet != null;
+		if (this.currentResultSet == null) {
+			return false;
+		}
+		boolean isOpen = false;
+		if (!this.currentResultSet.isClosed()) {
+			this.currentResultSet.close();
+			isOpen = true;
```

Comment:
tiny detail but I would put `isOpen = true` before the `close` call, as it is a bit strange-looking otherwise (someone could think "why is it open since I just closed it?")

Reply:
done!